File: backend-model/src/main/java/outcobra/server/model/dto/TeacherDto.java
package outcobra.server.model.dto;

import outcobra.server.model.Teacher;
import outcobra.server.model.interfaces.MappableDto;
import outcobra.server.model.interfaces.Mapper;

public class TeacherDto implements MappableDto<TeacherDto, Teacher> {
    @Override
    public Mapper<TeacherDto, Teacher> getMapper() {
        throw new UnsupportedOperationException("not implemented");
    }

    @Override
    public Teacher toEntity() {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: backend/src/main/kotlin/outcobra/server/model/dto/TeacherDto.kt
package outcobra.server.model.dto


import outcobra.server.model.domain.User
import outcobra.server.model.interfaces.OutcobraDto
import outcobra.server.model.interfaces.ParentLink

data class TeacherDto(val id: Long = 0,
                      val userId: Long = 0,
                      val name: String = "",
                      val email: String = "") : OutcobraDto {

    override val identifier: Long get() = id
    override val parentLink: ParentLink
        get() = ParentLink.make(userId, User::class.java)
}